code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1274,16 +1274,19 @@
 			item_flags[I] = I.flags
 			I.flags = 0 // Temporary set the flags to 0
 
+	if(!transformation) //Distinguish between creating a mob and switching species
+		dna.species.on_species_gain(src)
+
 	if(retain_damage)
 		//Create a list of body parts which are damaged by burn or brute and save them to apply after new organs are generated. First we just handle external organs.
 		var/bodypart_damages = list()
 		//Loop through all external organs and save the damage states for brute and burn
-		for(var/obj/item/organ/external/E in bodyparts)
-			if(E.brute_dam == 0 && E.burn_dam == 0 && E.internal_bleeding == FALSE) //If there's no damage we don't bother remembering it.
+		for(var/obj/item/organ/external/E as anything in bodyparts)
+			if(E.brute_dam == 0 && E.burn_dam == 0 && !(E.status & ORGAN_INT_BLEEDING)) //If there's no damage we don't bother remembering it.
 				continue
 			var/brute = E.brute_dam
 			var/burn = E.burn_dam
-			var/IB = E.internal_bleeding
+			var/IB = (E.status & ORGAN_INT_BLEEDING)
```

Comment:
Not something you've done, but for some reason this proc only tracks broken bones on *internal* organs and not external, so transforming into another species completely heals any fractures.
You should be able to fix that by just moving the checks onto the external loops since (as far as I know) it's impossible for internal organs to get broken in the first place.

Reply:
ORGAN_BROKEN on internal organs is used when they are necrotic. I was gonna make a separate PR to fix that at after this was merged.